src/hooks/useOpenApp.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react'
+
+/** Internals */
+
+enum AppAction {
+  CloseApp = 'closeApp',
+  OpenApp = 'openApp'
+}
+
+enum AppState {
+  Closed = '',
+  Opened = ' App--opened '
+}
+
+const openAppEvent = new Event(AppAction.OpenApp)
+
+const closeAppEvent = new Event(AppAction.CloseApp)
+
+/** API */
+
+export const openApp = (): boolean => window.dispatchEvent(openAppEvent)
+
+export const closeApp = (): boolean => window.dispatchEvent(closeAppEvent)
+
+export const useOpenApp = (): AppState => {
+  const [state, setState] = useState(AppState.Closed)
+
+  useEffect(() => {
+    window.addEventListener(AppAction.OpenApp, () => setState(AppState.Opened))
+    window.addEventListener(AppAction.CloseApp, () => setState(AppState.Closed))
+  }, [])
+
+  return state
```

Comment:
When calling `useOpenApp` I don't expect to retrieve a css class name. Maybe the method name should reflect that a bit more?

Or maybe we can use destructuration?

Something like `const { openAppCss } = useOpenApp()` ?

Reply:
I'm returning `{ getAppState: AppState }`
I think it's not too important to put CSS in the getter name because TS tells us what to expect.
Also putting CSS in the getter is too specific. By keeping that way, I can replace the string by a boolean if needed.
Also I wanted to return directly the classname because then, the App component has 0 logical computation to make, it just uses what I tell it to use.